src/goto-programs/goto_clean_expr.cpp

```diff
@@ -439,8 +443,8 @@ void goto_convertt::clean_expr_address_of(
   {
     DATA_INVARIANT(
       expr.operands().size() == 1, "ID_compound_literal has a single operand");
-    clean_expr(expr.op0(), dest, mode);
-    expr = make_compound_literal(expr.op0(), dest, mode);
+    clean_expr(to_unary_expr(expr).op(), dest, mode);
```

Comment:
Remove the `DATA_INVARIANT`. And I'd introduce `exprt op = to_unary_expr(expr).op();` and use that here and in the next instruction.

Reply:
done